mantidproject/mantid
# Apply ties in correct order

**Description of work.**
Fit uses a list of ties sorted in such a way that a tie is applied to a parameter before it appears in the right-hand side of another tie. The sorting procedure finds any circular dependencies.

**Report to:** nobody.

**To test:**
* Define a function and set complex ties where the same parameters appear both in the left and the right-hand sides. Do a fit and check that the tie equations are satisfied.
* Check that ties that have circular dependencies make Fit give an error. Examples: `A=A`, `A=B, B=A`, `A=B, B=C, C=A`

Fixes #22925

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
